lib/statsd/instrument.rb

```diff
@@ -151,9 +168,11 @@ def statsd_count_if(method, name, *metric_options)
     # @param name (see #statsd_measure)
     # @param metric_options (see #statsd_measure)
     # @return [void]
-    def statsd_count(method, name, *metric_options)
+    def statsd_count(method, name, *metric_option_args)
+      #metric_options = metric_options_args
       add_to_method(method, name, :count) do |old_method, new_method, metric_name|
         define_method(new_method) do |*args, &block|
+          metric_options = StatsD::Instrument.process_metric_options(metric_option_args, self, *args)
           StatsD.increment(StatsD::Instrument.generate_metric_name(metric_name, self, *args), 1, *metric_options)
```

Comment:
Can you move these two lines into an `ensure` block, so it runs after the original method has run? This brings it in line with the other methods.